---
layout: report
title: Seasonal influenza circulation patterns and future projections
linktitle: Sep 2016
categories:
 - reports
---

<div class="row">
	<div class="col-md-3">
		<div class="io-container title">
			A/H3N2
		</div>
		<div class="spacer"></div>
	</div>
	<div class="col-md-9">

		<b>Despite the late season 3c3.a epidemic in the USA, we predict clade 3c2.a viruses will continue to predominate in the H3N2 population. Within clade 3c2.a, the 171K variant has spread rapidly with the majority of recent H3N2 infections comprised of 171K viruses. Barring the emergence of new antigenic variant, we believe 171K will continue to predominate.</b>

		<div class="spacer"></div>

		<!-- Discussion of sampling -->

    <p>We base our primary analysis on a set of viruses collected between Sep 2014 and Aug 2016, comprising approximately 100 viruses per month where available and seeking to equilibriate sample counts geographically where possible. This equilibration attempts to collect equal samples from Africa, China, Europe, Japan/South Korea, North America, Oceania, South America, South Asia, Southeast Asia and West Asia. In the following analysis we collapse samples from China, South Asia, Southeast Asia, Japan and Korea into a single region that we're calling "Asia", resulting in Asia possessing greater sample counts than North America or Europe. The only month that significantly departs from equitable sampling is Aug 2016 with 38 viruses, primarily from Europe and North America. We subsample to 100 viruses per month and not more to keep sample counts as equitable as possible across space and time.

		{% embed_image /reports/figures/sep-2016/H3N2_counts.png 12 %}

		<!-- Discussion of older 3c3.a, 3c3.b and 3c2.a clades -->

    <p>Viral clades 3c3.a, 3c3.b and 3c2.a emerged from the Texas/2012 background in early 2014 and rapidly spread through the viral population. Subsequently, we have observed competition among these clades, with 3c2.a viruses being globally dominant beginning in 2015. Recently, we have observed the steady decline of 3c3.b viruses. At this point, we estimate that they are either extinct or nearly extinct. 3c3.a viruses were largely replaced by 3c2.a viruses starting in 2015. However, we have observed an anomolous late-season epidemic of 3c3.a viruses within the USA from Jan to Aug 2016. Elsewhere in the world, 3c2.a viruses have remained dominant, particularly in E Asia, where we estimate 3c2.a frequencies to be >95% throughout 2016.

		{% embed_image /reports/figures/sep-2016/H3N2_clades.png 12 %}

		<!-- Discussion of late season 3c2.a viruses in the USA -->

    <p>The late-season 3c3.a epidemic in the USA is puzzling. However, we suspect that this epidemic is due to epidemiologic circumstance rather than the emergence of a antigenically novel variant. The strongest evidence for this is that there is not a single clade within 3c3.a that is spreading throughout the USA. Instead, a variety of 3c3.a viruses are spreading, each with different HA1 mutations. The emergence and spread of an adaptive variant would have appeared as a single clade bearing a characteristic epitope mutation. This is not what we see. Because of this, we highly doubt that these 3c3.a viruses will spread globally. It is conceivable however, that the 2016-2017 USA epidemic could derive from over-summering transmission chains. This would be an unusual event, but not unheard of (<a href="http://bedford.io/papers/bedford-global-migration/">Bedford et al. 2010</a>). Still, we believe that 2016-2017 USA season will likely derive from 171K viruses due to their rapid spread (see below).

		{% embed_image /reports/figures/sep-2016/H3N2_3c3a_tree.png 12 %}

		<!-- Discussion of variation within 3c2.a -->

    <p>The dominant 3c2.a viruses have continued to genetically diversify with emergence of multiple subclades. We observe 3 subclades of decent frequency in 2016 viruses. These are characterized by HA1 mutations 142K/197R, HA1 mutation 197K and HA1 mutation 171K + HA2 mutations 77V/155E. The 171K variant comprises approximately 54% of 3c2.a viruses collected in 2016.

    {% embed_image /reports/figures/sep-2016/H3N2_3c2a_tree.png 12 %}

    <!-- Discussion of dynamics within 3c2.a and rise of 171K -->

    <p>The 142K/197R clade first emerged around June 2015 and rose to nearly 20% global frequency in Nov 2015. However, it's declined throughout 2016. We doubt this is a competitive virus. The 197K clade emerged in late 2015 and has slowly grown in frequency since. We estimate that it now comprises ~14% of H3N2 viruses. Notably however, 197K is at extremely low frequency in East Asia with an estimated present-day frequency ~2%. On the other hand, clade 171K viruses have done remarkdably well throughout 2016 and now comprise an estimated 69% of currently circulating H3N2 viruses. This steady increase is strongly suggestive of an adaptive origin. Notably, 171K emerged and spread first in Asia, reaching nearly 80% frequency in April 2016. Higher frequency of 171K in Asia is expected to spread to the rest of the world given past geographic observations (<a href="http://bedford.io/papers/bedford-global-circulation/">Bedford et al. 2015</a>).

    <p><i>Without strong competition from another novel H3N2 virus, we believe that 171K will continue to increase in frequency in the global population and predominant in the 2016-2017 influenza season.</i> The continued spread of 171K is fully in line with the predictions we made in <a href="/reports/feb-2016/">Feb 2016</a>.

    {% embed_image /reports/figures/sep-2016/H3N2_mutations.png 12 %}

    <!-- Discussion of LBI -->

    <p>Other indicators suggest evolutionary success of 171K viruses. Notably "local branching index" (<a href="https://elifesciences.org/content/3/e03568">Neher et al. 2014</a>) supports 171K as a high fitness virus. 3c3.a viruses and other clades within 3c2.a and do not show signal in the "local branching index" analysis.

    {% embed_image /reports/figures/sep-2016/H3N2_LBI.png 12 %}

		<!-- Discussion of antigenicity -->

		<p>Unfortuantely, we lack sufficient recent serological data to distinguish antigenic evolution within 3c2.a and 3c3.a clades. The previous observations derive entirely from genetic data.

		<!-- Discussion of basal viruses -->

		<p>There are a variety of viruses at the base of the 171K clade, possessing HA1 171K along with HA2 77V/155E, but lacking further amino acid changes in HA.

    {% embed_image /reports/figures/sep-2016/H3N2_171K_basal.png 12 %}

	</div>
</div>

<div class="row">
	<div class="col-md-12">
		<div class="bigspacer"></div>
	</div>
</div>

<div class="row">
	<div class="col-md-3">
		<div class="io-container title">
			A/H1N1pdm
		</div>
		<div class="spacer"></div>
	</div>
	<div class="col-md-9">

    <b>The clade 6b.1, comprising 84N/162N/216T, has continued to rise and predominate in the H1N1pdm population throughout 2016. Almost all circulating H1N1pdm viruses are now 6b.1. There is not yet obvious evolution within this clade.</b>

		<div class="spacer"></div>

		<!-- Discussion of sampling -->

		<p>We base our primary analysis on a set viruses collected between Sep 2014 and Aug 2016, comprising approximately 100 viruses per month where available and seeking to equilibriate sample counts geographically where possible. As discussed above, this equilibration attempts to collect equal samples from Africa, China, Europe, Japan/South Korea, North America, Oceania, South America, South Asia, Southeast Asia and West Asia. In the following analysis we collapse samples from China, South Asia, Southeast Asia, Japan and Korea into a single region that we're calling "Asia", resulting in Asia possessing greater sample counts than North America or Europe. Recent months through June 2016 have largely sufficient sample counts and sample distributions. There are fewer samples from July to present.

		{% embed_image /reports/figures/sep-2016/H1N1pdm_counts.png 12 %}

    <!-- Discussion of phylogenetic structure within clade 6b, define clades 6b.1 and 6b.2 -->

		<p>Within clade 6b, two major genetic variants have emerged. These are clade 6b.1 comprised of HA1 84N/162N/216T and clade 6b.2 comprised of HA1 152T and HA2 174E. Most recent samples have been from 6b.1 viruses, with 85% of 2016 samples being from 6b.1 viruses.

		{% embed_image /reports/figures/sep-2016/H1N1pdm_tree.png 12 %}

		<!-- Discussion of frequencies / dynamaics of 6b.1 and 6b.2 -->

    <p>At this point, all regions of the world are dominated by 6b.1 viruses. This clade rose from low frequency in Aug 2015 to reach present day global frequencies of ~98%. There remain a minority of circulating 6b.2 viruses. We estimate 2% of H1N1pdm viruses globally to be 6b.2. These are slightly higher prevalence in Asia, but still a distinct minority. We estimate that 6b.1 is at 85% frequency in Asia, while 6b.2 is at 15% in Asia. Notably, the frequency of 6b.2 has remained stable for almost 12 months.

    <p><i>Every indication suggests the continued dominance of 6b.1 viruses. We are now watching for the emergence of genetic variants within the 6b.1 clade.</i> Notably, the continued rise and dominance of 6b.1 viruses fits with predictions in <a href="/reports/feb-2016/">Feb 2016</a>.

		{% embed_image /reports/figures/sep-2016/H1N1pdm_clades.png 12 %}

  </div>
</div>
