---
layout: report
title: Seasonal influenza circulation patterns and projections for 2017-2018
linktitle: Feb 2017
categories:
 - reports
---

<div class="row">
    <div class="col-md-3">
        <div class="io-container title">
            A/H3N2
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">

        <b>Clade 3c2.a has continued to diversify genetically with a complicated and rapid dynamics of different subclades. This diversification is not reflected in serological data that shows only minor to moderate antigenic evolution. Nevertheless, the highly parallel mutation patterns and the rapid rise and fall of clades suggests a competitive dynamics of phenotypically different viruses.</b>

        <div class="spacer"></div>

        <!-- Discussion of sampling -->

    <p>We base our primary analysis on a set of viruses collected between Jan 2015 and Feb 2017. We analyze mutation frequency trajectories separately in different geographic regions using all available sequence data. Our phylogenetic analysis is based on a subsample of approximately 100 viruses per month where available and seeking to equilibrate sample counts geographically where possible. This equilibration attempts to collect equal samples from Africa, China, Europe, Japan/South Korea, North America, Oceania, South America, South Asia, Southeast Asia and West Asia. In the following analysis we collapse samples from China, South Asia, Southeast Asia, Japan and Korea into a single region referred to here as "Asia", resulting in Asia possessing greater sample counts than North America or Europe. Europe typically contributes few sample during Northern hemisphere summer, North America is well represented throughout. Dec 2016 and Jan 2017 are dominated by viruses sampled in Europe and North America.

        {% embed_image /reports/figures/feb-2017/h3n2_total_counts.png 9 %}
        {% embed_image /reports/figures/feb-2017/h3n2_tree_counts.png 9 %}

        <!-- Discussion of older 3c3.a, 3c3.b and 3c2.a clades -->

    <p>Of the clades that emerged from the Texas/2012 background in early 2014, only 3c2.a remains. The uptick of 3c3.a viruses in mid 2016 in US has disappeared and 3c3.b viruses have not been observed in substantial numbers for a year. Within clade 3c2.a, a number of subclades with signature mutation N171K, T131K/R142K, R142K, N121K/R144K, Q197K have emerged. The largest of these clades is the 171K clade which was first observed in mid 2015 and is now globally at a frequency of approximately 80%.

        {% embed_image /reports/figures/feb-2017/h3n2_frequencies.png 12 %}

        <!-- Discussion of subclades of 3c2.a -->

    Clades R142K, N121K/R144K, Q197K arose in late 2014 or early 2015 but have not come to dominate.
    Currently, the only noteworthy sister clade of 171K is the clade with mutations T131K/R142K.
    This clade was first observed mid-2016 in China and dominated in China during last 3 month according to sequences available in GISAID.
    It has been one of the most rapidly rising clades and has spread to Europe and North America.
    In the branch leading up to this clade, we observe a rapid succession of mutation in codon 131 from threonine to asparagine to lysine and the mutation R142K.
    Rapid succession of multiple mutations in the same codon suggest that these changes are adaptive.
    Interestingly, we observe a large number of mutations to positively charged residues and particularly many parallel mutations to lysine at positions 121, 131, 142, 144, 171, and 197.

        {% embed_image /reports/figures/feb-2017/h3n2_tree.png 12 %}

        <!-- Discussion of antigenic diversity -->
    Despite the large genetic diversity at sites in the vicinity of the receptor binding site, HI or FRA assays show no evidence of substantial antigenic change.
    HI assays by the WHO collaborating center at the Crick Institute in London indicate a moderate two-fold titer drop relative to A/HongKong/4801/2014 for several subclades, including the 171K subclade.
    However, there is substantial variability in those data.
    More dense and recent data by the CDC in Atlanta, Georgia, do not support a homogeneous titer drop of the 171K clade relative to the current vaccine strain Hong Kong/4801/2014.
    The figure below shows trees colored by predicted titer drop relative to Hong Kong/4801/2014 using the tree model trained on data by HI data by Crick and CDC as well as FRA data by CDC.
    Taken together, these data provide little evidence for substantial antigenic evolution within 3c2.a.

        {% embed_image /reports/figures/feb-2017/h3n2_tree_model.png 12 %}
        <!-- Discussion of predictive indices -->

    The local branching index -- LBI, a phylogenetic indicator of clade growth -- corroborates the observations made on the basis of clade frequencies. Over the last 6 month, the 131K clade has highest LBI, while the clade 171K has highest LBI when considering sequences in all of 2016.
    Interpretation of the sequence based indices such as number of epitope or non-epitope mutations is difficult due the extensive genetic heterogeneity. The clade 131K has accumulated the same number of epitope mutations as 171K but has fewer non-epitope mutations relative to A/Texas/50/2012. This might indicate that this clade is intrinsically fit, consistent with its rapid growth in China and global spread.

        {% embed_image /reports/figures/feb-2017/h3n2_lbi.png 8 %}


    </div>
    <div class="col-md-3">
        <div class="io-container title">
            A/H1N1pdm
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">
    <p><b>Very few H1N1pdm viruses have been observed in recent month. The dominant clade continues to be 6b.1 and there is little amino acid sequence variation in HA. The only notable subclade that has been rising recently is the clade with mutation HA1:R205K and HA1:S183P at its base. This clade is dominated by North American viruses and we see no evidence that this clade has a particular competitive advantage.</b></p>

        {% embed_image /reports/figures/feb-2017/h1n1pdm_total_counts.png 9 %}
        {% embed_image /reports/figures/feb-2017/h1n1pdm_tree_counts.png 9 %}

        <!--  -->

    <p>The graphs below illustrate the continued dominance of 6b.1 (mutation 162N). Mutation 183P has been rising in North America starting in early-mid 2016, but this rise corresponds to a small number of cases and might turn out insignificant during the next H1N1pdm dominated season. </p>

        {% embed_image /reports/figures/feb-2017/h1n1pdm_frequencies.png 12 %}

    Notably, however, there is a quick succession of three amino acid mutations R205K, S183P, I166V in HA1. No consistent antigenic variation is observed among recently circulating H1N1pdm viruses.

        {% embed_image /reports/figures/feb-2017/h1n1pdm_tree.png 12 %}

    </div>

    <div class="col-md-3">
        <div class="io-container title">
            B/Vic
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">
    <p><b>Clade 1A has continued to dominate and mutation 117V has all but taken over the global population. The rise of this mutation was fairly gradual and we have no evidence that it is associated with antigenic change or other benefit to the virus.</b></p>
        {% embed_image /reports/figures/feb-2017/vic_total_counts.png 9 %}
        {% embed_image /reports/figures/feb-2017/vic_tree_counts.png 9 %}

        <!--  -->

        {% embed_image /reports/figures/feb-2017/vic_frequencies.png 12 %}

    <p>Neither the local branching index of clade frequencies single out a particular variant within the 1A/I117V clade.</p>
        {% embed_image /reports/figures/feb-2017/vic_tree.png 12 %}
    </div>
    <div class="col-md-3">
        <div class="io-container title">
            B/Yam
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">
    <p><b>Calde 3 has continued to dominate. Within clade 3, a clade with mutation HA1:251V is globally at frequency of about 80% throughout 2016. Within this clade, mutation 211R is at 25% frequency. In addition, a clade without prominent amino acid mutations has been rising throughout 2016.</b></p>
        {% embed_image /reports/figures/feb-2017/yam_total_counts.png 9 %}
        {% embed_image /reports/figures/feb-2017/yam_tree_counts.png 9 %}

        <!-- Discussion of older 3c3.a, 3c3.b and 3c2.a clades -->


        {% embed_image /reports/figures/feb-2017/yam_frequencies.png 12 %}
        {% embed_image /reports/figures/feb-2017/yam_tree.png 12 %}

    <p>The LBI highlights the central clade in the tree below. This clade has not amino acid mutations in HA along its backbone.</p>
        {% embed_image /reports/figures/feb-2017/yam_lbi.png 8 %}

    </div>


</div>
